src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateForEachAction = { ...state };
```

Comment:
[NAMING] - The variable name `stateForEachAction` is not as descriptive as it could be. Consider renaming it to `currentState` or `stateCopy` to better reflect that it is a copy of the state that will be modified after each action.